cmd/test/test.go

```diff
@@ -0,0 +1,228 @@
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			flagNames := []string{
+				"all-namespaces",
+				"combine",
+				"data",
+				"fail-on-warn",
+				"ignore",
+				"namespace",
+				"no-color",
+				"no-fail",
+				"suppress-exceptions",
+				"output",
+				"parser",
+				"policy",
+				"proto-file-dirs",
+				"capabilities",
+				"trace",
+				"strict",
+				"update",
+				"junit-hide-message",
+				"quiet",
+			}
+			for _, name := range flagNames {
+				if err := viper.BindPFlag(name, cmd.Flags().Lookup(name)); err != nil {
+					return fmt.Errorf("bind flag: %w", err)
+				}
+			}
+
+			return nil
+		},
+
+		RunE: func(cmd *cobra.Command, fileList []string) error {
+			ctx := cmd.Context()
+
+			if len(fileList) < 1 {
+				cmd.Usage() //nolint
+				return fmt.Errorf("missing required arguments")
+			}
+
+			var runner runner.TestRunner
+			if err := viper.Unmarshal(&runner); err != nil {
+				return fmt.Errorf("unmarshal parameters: %w", err)
+			}
+
+			results, err := runner.Run(ctx, fileList)
+			if err != nil {
+				return fmt.Errorf("running test: %w", err)
+			}
+
+			var exitCode int
+			if runner.FailOnWarn {
+				exitCode = output.ExitCodeFailOnWarn(results)
+			} else {
+				exitCode = output.ExitCode(results)
+			}
+
+			if !runner.Quiet || exitCode != 0 {
+				if runner.Output == OutputAppstudio {
+					// The appstudio format is unknown to Conftest so we handle it ourselves
+					report := applicationsnapshot.AppstudioReportFromCheckResults(results, runner.Namespace)
+					reportOutput, err := json.Marshal(report)
+					if err != nil {
+						return fmt.Errorf("output results: %w", err)
+					}
+					fmt.Printf("%s\n", reportOutput)
+
+				} else {
+					// Conftest handles the output
+					outputter := output.Get(runner.Output, output.Options{
+						NoColor:            runner.NoColor,
+						SuppressExceptions: runner.SuppressExceptions,
+						Tracing:            runner.Trace,
+						JUnitHideMessage:   viper.GetBool("junit-hide-message"),
+					})
+					if err := outputter.Output(results); err != nil {
+						return fmt.Errorf("output results: %w", err)
+					}
+				}
+
+				// When the no-fail parameter is set, there is no need to figure out the error code
+				// as we always want to return zero.
+				if runner.NoFail {
+					return nil
+				}
+			}
+
+			os.Exit(exitCode)
+			return nil
+		},
+	}
+
+	cmd.Flags().Bool("fail-on-warn", false, "Return a non-zero exit code if warnings or errors are found")
+	cmd.Flags().Bool("no-fail", false, "Return an exit code of zero even if a policy fails")
+	cmd.Flags().Bool("no-color", false, "Disable color when printing")
+	cmd.Flags().Bool("suppress-exceptions", false, "Do not include exceptions in output")
+	cmd.Flags().Bool("all-namespaces", false, "Test policies found in all namespaces")
+	cmd.Flags().Bool("quiet", false, "Disable successful test output")
+
+	cmd.Flags().BoolP("trace", "", false, "Enable more verbose trace output for Rego queries")
```

Comment:
You're not specifying a single letter option here. Linked 203-205 can use `cmd.Flags().Bool` in that case.

Reply:
I agree, but this is verbatim upstream Conftest code, so the fix should be in conftest.

Reply:
Here we go: https://github.com/open-policy-agent/conftest/pull/847 :grin:

Ps, if our code diverges too much from the original code it becomes harder to maintain when/if the original code changes. So there's a tradeoff I think.

Reply:
Fixed in latest revision.